### PR TITLE
Scroller enum types renaming

### PR DESCRIPTION
**Pull request by RBrid**
Applying some of the changes that came out of the API review meetings:
- Renaming enumerations
    ScrollerScrollMode --> ScrollMode
    ScrollerZoomMode --> ZoomMode
    ScrollerChainingMode --> ChainingMode
    ScrollerRailingMode --> RailingMode
    ScrollerInputKind --> InputKind
- Removing the ScrollMode.Auto enum value and make ScrollMode.Enabled the new default. I am preserving the old code with #ifdef USE_SCROLLMODE_AUTO since the Auto value may be re-introduced later.

### CODE REVIEW

File: dev/ParallaxView/APITests/ParallaxViewTests.cs [hunk 1]
```diff
@@ -1433,7 +1433,7 @@ public void VerifyParallaxViewOffsetChangeInsideScrollViewer()
             scrollViewer.Height = c_defaultUIScrollViewerHeight;
             scrollViewer.HorizontalScrollBarVisibility = ScrollBarVisibility.Hidden;
             scrollViewer.VerticalScrollBarVisibility = ScrollBarVisibility.Hidden;
-            scrollViewer.ZoomMode = ZoomMode.Disabled;
+            scrollViewer.ZoomMode = Windows.UI.Xaml.Controls.ZoomMode.Disabled;
```
Comment (ranjeshj) [inline dev/ParallaxView/APITests/ParallaxViewTests.cs]: nit: fully qualified here but not below.

File: dev/Scroller/Scroller.idl [hunk 2]
```diff
@@ -8,26 +8,32 @@ enum InteractionState
...
+enum ScrollMode
 {
-    Auto = 0,
-    Enabled = 1,
-    Disabled = 2,
+    Enabled = 0,
+    Disabled = 1,
+#ifdef USE_SCROLLMODE_AUTO
+    Auto = 2,
+#endif
```
Comment (ranjeshj) [inline dev/Scroller/Scroller.idl]: Just curious, what is the motivation for removing auto ? How do i show the scrollbars only if the content is bigger than the viewport ?

File: dev/ParallaxView/APITests/ParallaxViewTests.cs (hunk 1 above)
Comment (RBrid) [reply dev/ParallaxView/APITests/ParallaxViewTests.cs]: This was done to disambiguate between the WUX and MUX versions of the enum.

File: dev/Scroller/Scroller.idl (hunk 2 above)
Comment (RBrid) [reply dev/Scroller/Scroller.idl]: The motivation was to give us the option to introduce a property in the future that governs the overpan behavior in general. ScrollMode.Auto is about the ability to overpan or not when the content is smaller than the viewport.
In order to show the scrollbar only when the content is larger than the viewport, use the ScrollBarVisibility.Auto value.

File: dev/ParallaxView/ScrollInputHelper.h [hunk 3]
```diff
@@ -34,7 +34,7 @@ class ScrollInputHelper
     winrt::UIElement GetScrollContentElement() const;
     winrt::HorizontalAlignment GetEffectiveHorizontalAlignment() const;
     winrt::VerticalAlignment GetEffectiveVerticalAlignment() const;
-    winrt::ZoomMode GetEffectiveZoomMode() const;
+    winrt::FxZoomMode GetEffectiveZoomMode() const;
```
[the remark this answers is not in the document]
Comment (RBrid) [reply dev/ParallaxView/ScrollInputHelper.h]: It's a shortcut for the WUX version of the enum, like FxScrollViewer is a shortcut for the WUX version of the control.  FxZoomMode maps to Windows.UI.Xaml.Controls.ZoomMode.

**Pull request merged**